src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -169,6 +169,15 @@ private void OnImeOptionsChanged(ImeAction imeAction)
 					using (focusState == FocusState.Programmatic ? PreventKeyboardDisplayIfSet() : null)
 					{
 						_textBoxView.RequestFocus();
+
+						var selectionStart = this.SelectionStart;
+
+						if (selectionStart == 0)
+						{
+							int cursorPosition = selectionStart + _textBoxView?.Text?.Length ?? 0;
+
+							this.Select(cursorPosition, 0);
+						}
```

Comment:
This is very confusing. Why are we altering the selection here? I think Android didn't have a bug in the first place.

I'm not sure, but it seems you're changing Android to match Wasm behavior instead of the opposite? The platform that had a bug, IIRC, was Wasm only.

Reply:
The test fail because the Android does not mimic the UWP behavior, then I fixed that.